### PR TITLE
Allow to scan more than one kind of bar code at once

### PR DESCRIPTION
**Pull request by fbochu**
As suggested by @AndreMiras, zbarlight should be able to scan more than
one code type at once.

Linked to #23

### CODE REVIEW

File: src/zbarlight/__init__.py [hunk 1]
```diff
@@ -39,22 +39,34 @@ def scan_codes(code_type, image):
...
+
+    # Translate symbologies
+    symbologies = [
+        Symbologies.get(code_type.upper())
+        for code_type in set(code_types)
+    ]
+
+    # Check that all symbologies are known
+    if None in symbologies:
+        bad_code_types = [code_type for code_type in code_types if code_type not in Symbologies]
```
Comment (dbaty) [inline src/zbarlight/__init__.py]: You use `code_type.upper()` as the key in `Symbologies` above when defining `symbologies`, but here you look for `code_type`. As such, the following line could wrongly says that a code type is unknown if it's provided in lower case, if there is also an unknown code type.

Proposal:

    symbologies = []
    for code_type in set(code_types):
        try:
            symbologies.append(Symbologies[code_type.upper()])
        except KeyError:
            raise UnknownSymbologieError('Unknown symbology:  %s' % code_type)

This version raises when it finds the first wrong code type (and does not mention all wrong types as your version does), but I think it's good enough. Alternative:

    symbologies = []
    unknown = []
    for code_type in set(code_types):
        try:
            symbologies.append(Symbologies[code_type.upper()])
        except KeyError:
            unknown.append(code_type)
    if unknown:
        raise UnknownSymbologieError('Unknown symbologies:  %s' % ', '.join(unknown))

File: src/zbarlight/__init__.py [hunk 2]
```diff
@@ -39,22 +40,39 @@ def scan_codes(code_type, image):
...
+        code_types (list(str)): Code type(s) to search (see ``zbarlight.Symbologies`` for supported values).
         image (PIL.Image.Image): Image to scan
 
     returns:
         A list of *code_type* code values or None
 
     """
+    if isinstance(code_types, str):
+        code_types = [code_types]
+        warnings.warn(
```
Comment (AndreMiras) [inline src/zbarlight/__init__.py]: Well done!

**Comment by AndreMiras** [comment]
Nice pull request I have to say, nice coverage and everything, thank you!

**Comment by AndreMiras** [comment]
OK nice company :+1: 

image

I wanted to add the following note, but I failed:
> Thanks @fbochu @Polyconseil @PSF https://github.com/Polyconseil/zbarlight/pull/24#issuecomment-394336432

Thanks again!

**Pull request merged**